networkdb/networkdb.go

```diff
@@ -477,7 +530,13 @@ func (nDB *NetworkDB) JoinNetwork(nid string) error {
 		nodeNetworks = make(map[string]*network)
 		nDB.networks[nDB.config.NodeName] = nodeNetworks
 	}
-	nodeNetworks[nid] = &network{id: nid, ltime: ltime}
+	nDB.networkEpoch[nid]++
```

Comment:
maybe I'm missing something, but if the reapTime is proper should guarantee that once the network is deleted by the other nodes there is no more messages going around for that network. If that is the case then, if the current node does not have that network object locally means that can start again from the networkEpoch of 1. All the other nodes won't have context to check it with so should be safe and we can remove the networkEpoch map completely.
Other approach can be to have only 1 integer that is used to seed all the epochs in an atomically incremental way just to guarantee that there is not overlap of epoch but still will be better than having to deal with the map